src/hooks/useCreateTarget.js

```diff
@@ -0,0 +1,67 @@
+import { useState, useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { Alert } from 'react-native';
+
+import { createTarget } from 'actions/targetActions';
+import { TARGETS_MAX } from 'constants/data';
+import useGetTargets from 'hooks/useGetTargets';
+import strings from 'localization';
+
+const latitudeDelta = 0.05;
+const longitudeDelta = 0.05;
+
+const useCreateTarget = (targets, mapCenter, setMapCenter) => {
+  const [newTarget, setNewTarget] = useState(null);
+  const [createFormVisible, setCreateFormVisible] = useState(false);
+
+  const { targetsRequest } = useGetTargets();
+
+  const dispatch = useDispatch();
+  const createTargetRequest = useCallback(target => dispatch(createTarget(target)), [dispatch]);
+
+  const changeTargetPosition = e => {
+    const { latitude, longitude } = e.nativeEvent?.coordinate;
```

Comment:
this could throw an error 
```suggestion
    const { latitude, longitude } = e.nativeEvent?.coordinate || {};
```